obsidian.el

```diff
@@ -529,7 +529,7 @@ link name must be available via `match-string'."
   "Add extension to wiki link F if none."
   (if (file-name-extension f)
       f
-    (s-concat f ".md")))
+    (s-concat (obsidian--remove-section f) ".md")))

```

Comment:
Any reason to not just use `replace-regexp-in-string` here instead of creating a new function?

`(s-concat (replace-regexp-in-string "#.*$" "" f) ".md")))`

Reply:
Originally I thought I will have to add the function call to multiple locations, so the function itself is not needed.
But I'd consider replacing `defun` with `defsubst` and keep it, as it becomes inline (good for performance),
and still tells what the purpose of the regex is (good for readability).